subiquity/server/controllers/shutdown.py

```diff
@@ -83,6 +83,20 @@ async def _run(self):
         elif self.app.state == ApplicationState.DONE:
             await self.shutdown()
 
+    async def copy_cloud_init_logs(self, target_logs):
+        # Preserve ephemeral boot cloud-init logs if applicable
+        cloudinit_logs = (
+            "/var/log/cloud-init.log",
+            "/var/log/cloud-init-output.log",
+        )
+        for logfile in cloudinit_logs:
+            if not os.path.exists(logfile):
+                continue
+            set_log_perms(logfile)
+            await self.app.command_runner.run(
+                ["cp", "-a", logfile, "/var/log/installer"]
```

Comment:
Why change the owner of the original as well?

Reply:
Simplicity.  We're in the shutdown controller and we are about to shutdown, we're doing the final final steps before unmount of /target.  The scenario where the ephemeral system log files are presumed to not matter, but maybe there is an obscure case where that is still a thing.